src/common/plugin/makeCurrencyTools.ts

```diff
@@ -30,36 +33,32 @@ export function makeCurrencyTools(
 ): EdgeCurrencyTools {
   const { currencyInfo, engineInfo } = pluginInfo
 
+  const asCurrencyPrivateKey = asPrivateKey(
+    engineInfo.network,
+    engineInfo.coinType
+  )
+  const wasCurrencyPrivateKey = uncleaner(asCurrencyPrivateKey)
+  const asCurrencyNumbWalletInfo = asNumbWalletInfo(engineInfo)
+
   const fns: EdgeCurrencyTools = {
     async createPrivateKey(
-      walletType: string,
+      _walletType: string,
       opts?: JsonObject
     ): Promise<JsonObject> {
-      const mnemonicKey = pluginUtils.getMnemonicKey({
-        coin: engineInfo.network
-      })
       const mnemonic = bip39.entropyToMnemonic(Buffer.from(io.random(32)))
-      const keys: JsonObject = {
-        [mnemonicKey]: mnemonic
-      }
 
-      return {
-        ...keys,
+      const privateKey = asCurrencyPrivateKey({
```

Comment:
Use a type definition here, instead of a cleaner:

```
const privateKey: PrivateKey = {
  privateKey: mnemonic,
  format: opts?.format ?? engineInfo.formats?.[0] ?? 'bip44',
  coinType: opts?.coinType ?? engineInfo.coinType ?? 0
}
```